scripts/limonade/src/index.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Connection,
+  GetProgramAccountsFilter,
+  PublicKey,
+} from "@solana/web3.js";
+import { limoId, Order } from "@kamino-finance/limo-sdk";
+import bs58 from "bs58";
+import { hideBin } from "yargs/helpers";
+import yargs from "yargs";
+import { Client, OpportunityCreate } from "@pythnetwork/express-relay-js";
+
+const argv = yargs(hideBin(process.argv))
+  .option("endpoint", {
+    description:
+      "Express relay endpoint. e.g: https://per-staging.dourolabs.app/",
+    type: "string",
+    default: "https://pyth-express-relay-mainnet.asymmetric.re/",
+  })
+  .option("rpc-endpoint", {
+    description:
+      "Solana rpc endpoint to use for subscribing to program account changes.",
+    type: "string",
+    default: "http://127.0.0.1:8899",
+  })
+  .option("global-config", {
+    description: "Global config of limo program to filter for.",
+    type: "string",
+    demandOption: true,
+  })
+  .option("chain-id", {
+    description: "Chain id to publish these orders for as opportunity",
+    type: "string",
+    demandOption: true,
+  })
+  .option("api-key", {
+    description:
+      "API key to authenticate with the express relay server for publishing opportunities.",
+    type: "string",
+    demandOption: true,
+  })
+  .help()
+  .alias("help", "h")
+  .parseSync();
+
+async function run() {
+  const connection = new Connection(argv.rpcEndpoint);
+
+  const globalConfig = new PublicKey(argv.globalConfig);
+  let filters: GetProgramAccountsFilter[] = [
+    {
+      memcmp: {
+        bytes: globalConfig.toBase58(),
+        offset: 8,
+      },
+    },
+    {
+      memcmp: {
+        bytes: bs58.encode(Order.discriminator),
+        offset: 0,
+      },
+    },
+  ];
+
+  console.log("Listening for program account changes");
+  const client = new Client({ baseUrl: argv.endpoint, apiKey: argv.apiKey });
+  let { blockhash: latestBlockhash } = await connection.getLatestBlockhash(
+    "confirmed"
+  );
+  const submitExistingOpportunities = async () => {
+    const response = await connection.getProgramAccounts(limoId, {
+      commitment: "confirmed",
```

Comment:
food for thought, use "processed"? a low percentage of processed blocks don't get confirmed ([supposedly ~5%](https://solana.com/docs/advanced/confirmation)), and this could improve latency of publishing opportunities if that's a source of delay. ofc there's a tradeoff, but in worst case the searcher's trade attempt fails on-chain.

Reply:
sorry this comment is more relevant for the filter in `onProgramAccountChange`